test/spec/app_spec.rb

```diff
@@ -585,28 +580,34 @@
 
     it "should add and remove source languages", bin6: true  do
       api_create_team_project_and_source_and_redirect_to_source('GOT', 'https://twitter.com/GameOfThrones')
-      sleep 5
-      element = @driver.find_element(:class, "source__edit-button")
+      wait_for_selector("source__tab-button-account",:class)
+      element = wait_for_selector("source__edit-button",:class)
       element.click
-      sleep 1
-      @driver.find_element(:class, "source__edit-addinfo-button").click
-      sleep 1
-      @driver.find_element(:class, "source__add-languages").click
+      wait_for_selector("source__edit-buttons-cancel-save",:class)
+      element = wait_for_selector("source__edit-addinfo-button",:class)
+      element.click
+      sleep 2
+      element = wait_for_selector("source__add-languages",:class)
+      element.click
+      sleep 2
       fill_field("sourceLanguageInput", "Acoli", :id)
       @driver.action.send_keys(:down).perform
       @driver.action.send_keys(:return).perform
-      sleep 2
+      sleep 1
       @driver.navigate.refresh
-      sleep 3
+      sleep 2
+      wait_for_size_change(0, "sourceLanguageInput",:id)
+      wait_for_selector("source__tab-button-media",:class)      
       expect(@driver.page_source.include?('Acoli')).to be(true)
```

Comment:
Shouldn't this be asserted before the `@driver.navigate.refresh` in order to be sure that auto-refresh worked?

Reply:
I think the result is the same, `wait_for_size_change(0, "sourceLanguageInput",:id)` being before or after the refresh. But I'll change.

The `wait_for_selector("source__tab-button-media",:class) ` needs to be after, it ensures that the page is completely loaded.

Reply:
I mean the line `expect(@driver.page_source.include?('Acoli')).to be(true)`. You're checking after the `@driver.navigate.refresh`. We need to be sure that the language is added after the user presses `ENTER`. The user shouldn't need to refresh the whole page to see the new language he/she added.